### PR TITLE
Human APP backend - 4th milestone

### PR DESCRIPTION
**Pull request by BartoszSolkaBD**
#### H-Captcha labeling enpoints:
    /enable - proxy to Reputation Oracle
    /verify - an endpoint that sends the user solution of the job to Captcha
    /daily-hmt-spent - an endpoint that returns global daily HMT spent (will be cached in Redis)
    /user-stats - an endpoint that returns stats per user (will be cached in Redis)

    
    
#### Proxy enpoints in reputation oracle: 
    /user/register-address
    /auth/web3/signin
    /assignment/resign-job
    
    
## Summary of changes

- hCaptcha
- proxy endpoints to reputation oracle 
- proxy endpoints to exchange oracle
- improved test coverage

## How test the changes
There are unit tests for all the features. To run them please execute `yarn test`

### CODE REVIEW

File: packages/apps/human-app/server/src/integrations/exchange-oracle/exchange-oracle.gateway.ts [hunk 1]
```diff
@@ -2,128 +2,162 @@ import { Injectable } from '@nestjs/common';
...
 import { Mapper } from '@automapper/core';
 import { InjectMapper } from '@automapper/nestjs';
-import { instanceToPlain } from 'class-transformer';
 import { HttpMethod } from '../../common/enums/http-method';
+import { toCleanObjParams } from '../../common/utils/gateway-common.utils';
+import { KvStoreGateway } from '../kv-store/kv-store.gateway';
+import { EscrowUtilsGateway } from '../escrow/escrow-utils-gateway.service';
 
 @Injectable()
 export class ExchangeOracleGateway {
```
Comment (portuu3) [inline packages/apps/human-app/server/src/integrations/exchange-oracle/exchange-oracle.gateway.ts:39]: Why ExchangeOracleGateway and ReputationOracleGateway don't have the same structure? For example use getEndpointOptions

File: packages/apps/human-app/server/src/integrations/h-captcha-labeling/h-captcha-statistics.gateway.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
+import { Injectable } from '@nestjs/common';
+import {
+  GatewayConfig,
+  GatewayEndpointConfig,
+} from '../../common/interfaces/endpoint.interface';
+import { HttpService } from '@nestjs/axios';
+import { GatewayConfigService } from '../../common/config/gateway-config.service';
+import { InjectMapper } from '@automapper/nestjs';
+import { Mapper } from '@automapper/core';
```
Comment (portuu3) [inline packages/apps/human-app/server/src/integrations/h-captcha-labeling/h-captcha-statistics.gateway.ts:9]:
```suggestion
```

File: packages/apps/human-app/server/src/modules/h-captcha/h-captcha.service.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    command: EnableLabelingCommand,
+  ): Promise<EnableLabelingResponse> {
+    return this.reputationOracleGateway.approveUserAsLabeler(command);
+  }
+
+  async getDailyHmtSpent(
+    command: DailyHmtSpentCommand,
+  ): Promise<DailyHmtSpentResponse> {
+    this.checkIfHcaptchaSitekeyPresent(command.siteKey);
+    const hmtKey = this.configService.dailyHmtSpentKey;
```
Comment (portuu3) [inline packages/apps/human-app/server/src/modules/h-captcha/h-captcha.service.ts:73]: Why is passed in an env?

**Review by portuu3 — CHANGES_REQUESTED** [review]
Rebase to develop branch.
Not tested yet

File: packages/apps/human-app/server/src/modules/statistics/statistics.service.ts [hunk 4]
```diff
@@ -55,14 +46,8 @@ export class StatisticsService {
     if (cachedStatistics) {
```
[the remark this answers is not in the document]
Comment (macnablocky) [reply packages/apps/human-app/server/src/modules/statistics/statistics.service.ts:46]: I will change this to make key from email and oracleAddress

File: packages/apps/human-app/server/src/modules/user-operator/model/operator.model.ts
[the remark this answers is not in the document]
Comment (macnablocky) [reply packages/apps/human-app/server/src/modules/user-operator/model/operator.model.ts:1]: ack, will divide this

File: packages/apps/reputation-oracle/server/vercel.json
[the remark this answers is not in the document]
Comment (macnablocky) [reply packages/apps/reputation-oracle/server/vercel.json:1]: ack, will rollback

File: packages/sdk/python/human-protocol-sdk/test/human_protocol_sdk/transaction/test_transaction_utils.py
[the remark this answers is not in the document]
Comment (macnablocky) [reply packages/sdk/python/human-protocol-sdk/test/human_protocol_sdk/transaction/test_transaction_utils.py:1]: ack, will rollback

File: packages/apps/human-app/server/src/modules/statistics/statistics.service.ts [hunk 5]
```diff
@@ -32,13 +28,8 @@ export class StatisticsService {
...
-    const exchangeOracleUrl =
-      await this.kvStoreGateway.getExchangeOracleUrlByAddress(command.oracleAddress);
-    const details = {
-      exchangeOracleUrl: exchangeOracleUrl,
-    } as OracleStatisticsDetails;
     const response: OracleStatisticsResponse =
-      await this.exchangeOracleGateway.fetchOracleStatistics(details);
+      await this.exchangeOracleGateway.fetchOracleStatistics(command);
     await this.cacheManager.set(
       url,
```
[the remark this answers is not in the document]
Comment (macnablocky) [reply packages/apps/human-app/server/src/modules/statistics/statistics.service.ts:34]: will change to address

File: packages/apps/human-app/server/src/integrations/h-captcha-labeling/h-captcha-statistics.gateway.ts (hunk 2 above)
Comment (macnablocky) [reply packages/apps/human-app/server/src/integrations/h-captcha-labeling/h-captcha-statistics.gateway.ts:9]: will apply

File: packages/apps/human-app/server/src/integrations/exchange-oracle/exchange-oracle.gateway.ts (hunk 1 above)
Comment (macnablocky) [reply packages/apps/human-app/server/src/integrations/exchange-oracle/exchange-oracle.gateway.ts:39]: Url for exchange oracle is received dynamically and in different ways for different endpoints (see postNewJobAssignment vs resignAssignedJob), as opposed to ones set for h-captcha/reputation oracle gateways - so we decided against alignment with other gateways

File: packages/apps/human-app/server/src/common/interfaces/endpoint.interface.ts [hunk 6]
```diff
@@ -1,3 +1,7 @@
+import {
+  HCaptchaLabelingStatsEndpoints,
+  ReputationOracleEndpoints,
+} from '../enums/reputation-oracle-endpoints';
```
[the remark this answers is not in the document]
Comment (macnablocky) [reply packages/apps/human-app/server/src/common/interfaces/endpoint.interface.ts:4]: will apply

File: packages/apps/human-app/server/src/common/config/gateway-config.service.ts [hunk 7]
```diff
@@ -68,7 +75,59 @@ export class GatewayConfigService {
               method: HttpMethod.POST,
               headers: this.JSON_HEADER,
             },
-          },
+            [ReputationOracleEndpoints.ENABLE_LABELING]: {
+              endpoint: '/user/register-labeler',
+              method: HttpMethod.POST,
+              params: {
+                api_key: this.envConfig.hcaptchaLabelingApiKey,
```
[the remark this answers is not in the document]
Comment (macnablocky) [reply packages/apps/human-app/server/src/common/config/gateway-config.service.ts:82]: ack, will remove and align controller

File: packages/apps/human-app/server/src/common/interfaces/jwt-token.model.ts
[the remark this answers is not in the document]
Comment (macnablocky) [reply packages/apps/human-app/server/src/common/interfaces/jwt-token.model.ts:1]: will move to common/utils

File: packages/apps/human-app/server/src/modules/statistics/statistics.service.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (portuu3) [reply packages/apps/human-app/server/src/modules/statistics/statistics.service.ts:46]: why dont you just use users wallet address?

[the remark this answers is not in the document]
Comment (macnablocky) [reply packages/apps/human-app/server/src/modules/statistics/statistics.service.ts:46]: Ok

**Review by portuu3 — APPROVED** [review]

**Pull request merged**